core-processor/src/ext.rs

```diff
@@ -471,40 +480,51 @@ impl CountersOwner for Ext {
     }
 
     fn gas_left(&self) -> GasLeft {
-        GasLeft {
-            gas: self.context.gas_counter.left(),
-            allowance: self.context.gas_allowance_counter.left(),
-        }
+        (
+            self.context.gas_counter.left(),
+            self.context.gas_allowance_counter.left(),
+        )
+            .into()
     }
 
-    fn set_gas_left(&mut self, gas_left: GasLeft) {
-        let GasLeft { gas, allowance } = gas_left;
+    fn current_counter(&self) -> CounterType {
+        self.current_counter
+    }
 
-        let gas_left = self.context.gas_counter.left();
-        if gas_left > gas {
-            if self.context.gas_counter.charge_if_enough(gas_left - gas) != ChargeResult::Enough {
-                // We checked above that `gas_left` is bigger than `gas`
-                unreachable!("Cannot charge {gas} from `gas_counter`");
-            }
-        } else {
-            self.context.gas_counter.refund(gas - gas_left);
+    fn decrease_to(&mut self, amount: u64) {
+        // For possible cases of non-atomic charges on backend side when global
+        // value is less than appropriate at the backend.
+        if self.current_val() <= amount {
+            log::trace!("Skipped decrease to global value");
+            return;
         }
```

Comment:
Is it a possible situation? Whether we really need to do nothing in the case ?

Reply:
Described case in comments